02-crud.c

```diff
@@ -0,0 +1,208 @@
+    }
+
+    fclose(file);
+}
+
+void updateFileStatus(bool found, const char *searchId, const char *msg)
+{
+    if (found)
+    {
+        remove(FILENAME);
+        rename("temp.txt", FILENAME);
+        printf("%s", msg);
+    }
+    else
+    {
+        remove("temp.txt");
+        printf("User with ID %s not found.\n", searchId);
+    }
+}
+bool updateUser(FILE *file, FILE *tempFile, const char *searchId)
+{
+    // A tempFile is best practice , because if we directly overWrite main file and if there is any program crash we lost our data.
+    User user;
+    bool found = false;
+    char choice = '\0';
+    while (fscanf(file, "%s %s %d", user.userId, user.name, &user.age) != EOF)
+    {
+        if (strcmp(user.userId, searchId) == 0)
+        {
+            printf("Do you want to update the name? (y/n): ");
+            getchar();
+            choice = getchar();
+            getchar();
+            if (choice == 'y' || choice == 'Y')
+            {
+                found = true;
+                printf("Enter new name (current: %s): ", user.name);
+                fgets(user.name, MAX_NAME_LEN, stdin);
+                user.name[strcspn(user.name, "\n")] = '\0';
+            }
+
+            // Ask if the user wants to update the age
+            printf("Do you want to update the age? (y/n): ");
+            choice = getchar();
+            getchar();
+            if (choice == 'y' || choice == 'Y')
+            {
+                found = true;
+                printf("Enter new age (current: %d): ", user.age);
+                scanf("%d", &user.age);
+            }
+        }
+        fprintf(tempFile, "%s %s %d\n", user.userId, user.name, user.age);
+    }
+
+    return found;
+}
+
+bool deleteUser(FILE *file, FILE *tempFile, const char *searchId)
+{
+    User user;
+    bool found = false;
+    while (fscanf(file, "%s %s %d", user.userId, user.name, &user.age) != EOF)
+    {
+        if (strcmp(user.userId, searchId) == 0)
+        {
+            found = true;
+            // we skip to  write this user into our file
+            continue;
+        }
+        fprintf(tempFile, "%s %s %d\n", user.userId, user.name, user.age);
+    }
+    return found;
+}
+
+void helperModify(int choice)
+{
+    FILE *file = fopen(FILENAME, "r"), *tempFile = fopen("temp.txt", "w");
+    if (!file)
+    {
+        printf("Unable to open the file.\n");
+        return;
+    }
+    if (!tempFile)
+    {
+        printf("Error while creating temporary file.\n");
+        fclose(file);
+        return;
+    }
+
+    char searchId[20];
+    printf("Enter the ID of the user to %s: ", (choice == 3) ? "update: " : "delete: ");
+    scanf("%s", searchId);
+
+    bool found = (choice == 3 ? updateUser(file, tempFile, searchId) : deleteUser(file, tempFile, searchId));
+
+    fclose(file);
+    fclose(tempFile);
+
+    updateFileStatus(found, searchId, (choice == 3 ? "User updated successfully!\n" : "User deleted successfully!\n"));
+}
+
+int main()
```

Comment:
In main() function, there should be only function calling.
There should not be things like "taking inputs", etc.